src/Microsoft.DotNet.ProjectJsonMigration/MigrationBackupPlan.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Internal.ProjectModel.Utilities;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    internal class MigrationBackupPlan
+    {
+        private readonly FileInfo globalJson;
+
+        public MigrationBackupPlan(
+            DirectoryInfo projectDirectory,
+            DirectoryInfo workspaceDirectory,
+            Func<DirectoryInfo, IEnumerable<FileInfo>> getFiles = null)
+        {
+            if (projectDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(projectDirectory));
+            }
+            if (workspaceDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(workspaceDirectory));
+            }
+
+            globalJson = new FileInfo(Path.Combine(
+                           workspaceDirectory.FullName,
+                           "global.json"));
+
+            projectDirectory = new DirectoryInfo(projectDirectory.FullName.EnsureTrailingSlash());
```

Comment:
Ah. I'm normalizing the input since there's no guarantee that the caller created it with a trailing slash. `DirectoryInfo` doesn't enforce it and its presence or absence will be visible in the `FullName` property.